### PR TITLE
Utils and stuffs added. My first testcase #4 for code review

### PR DESCRIPTION
**Pull request by andrej-khatkevich**
Some test cases added.

### CODE REVIEW

**Comment by andrej-khatkevich** [comment]
Found a way to replace Thread.sleep()

File: RubberDucksTest/src/test/java/mavenizer/staticPO/CataloguePage.java [hunk 1]
```diff
@@ -1,18 +1,68 @@
...
+        return driver.findElement(LocatorHelper.getLocator("CataloguePage.unitPrice")).getText();
+    }
+
+    @Step("Navigate to bin page")
+    public static void goToCartPage(WebDriver driver) {
         driver.findElement(LocatorHelper.getLocator("CataloguePage.cartLink")).click();
     }
 
+    @Step("Add some amount some category of ducks")
+    public static void addDucksToCart(WebDriver driver, Ducks duckType, String ducksQuantity) {
```
Comment (Fulcrum81) [inline RubberDucksTest/src/test/java/mavenizer/staticPO/CataloguePage.java:34]: since it's a number better use int, rather than string

**Review by Fulcrum81 — CHANGES_REQUESTED** [review]

File: RubberDucksTest/src/test/java/BinTest.java [hunk 2]
```diff
@@ -1,13 +1,73 @@
...
+import java.util.List;
+
+import static mavenizer.helpers.Ducks.*;
+
 public class BinTest extends TestBase {
+    /**
+     * Test case #4
+     */
     @Test
-    public void cartAmountTestNoMatterOfCurrency() {
```
[the remark this answers is not in the document]
Comment (andrej-khatkevich) [reply RubberDucksTest/src/test/java/BinTest.java:8]: Exactly. This was an example of where to place tests for our team

File: RubberDucksTest/src/test/java/BinTest.java [hunk 3]
```diff
@@ -1,13 +1,73 @@
...
-        Assert.assertEquals(CataloguePage.getCartAmountOnRightTopCorner(driver).replace("$", "").
-                replace("€", "").trim(), "0");
+    public void serialRemoveAllCategoriesTest() {
+        int categoriesInBinCount = 0;
+
+        CataloguePage.addDucksToCart(driver, PURPLE, "3");
+        categoriesInBinCount++;
```
[the remark this answers is not in the document]
Comment (andrej-khatkevich) [reply RubberDucksTest/src/test/java/BinTest.java:29]: You are absolutely right. My fail. Fixed

File: RubberDucksTest/src/test/java/BinTest.java [hunk 4]
```diff
@@ -1,13 +1,114 @@
...
+import java.time.Duration;
+import java.util.List;
+
+import static mavenizer.helpers.Ducks.*;
+
 public class BinTest extends TestBase {
+    /**
+     * Test case #4
+     */
+    @Description("Method adds categories(3) of ducks(3) to bin. Then removes it step by step " +
```
[the remark this answers is not in the document]
Comment (andrej-khatkevich) [reply RubberDucksTest/src/test/java/BinTest.java:22]: Fixed

File: RubberDucksTest/src/test/java/BinTest.java [hunk 5]
```diff
@@ -1,13 +1,114 @@
...
+        Assert.assertEquals(shortCutsLinksList.size(), categoriesInBinCount);
+
+        for (int i = categoriesInBinCount; i > 0; i--) {
+            if (shortCutsLinksList.size() > 1) { //Click all shortcuts except last one cause shortcut link absence
+                shortCutsLinksList.get(i - 1).click();
+                Assert.assertEquals(shortCutsLinksList.size(), i);
+                Assert.assertEquals(CartPage.getOrderSummaryTable(driver).getOrderSummaryRecords().size(), i);
+            }
+
+            new WebDriverWait(driver, Duration.ofSeconds(2))
```
[the remark this answers is not in the document]
Comment (andrej-khatkevich) [reply RubberDucksTest/src/test/java/BinTest.java:48]: Moved to Waits class as static method

File: RubberDucksTest/src/test/java/BinTest.java [hunk 6]
```diff
@@ -1,13 +1,114 @@
...
+        int initialAmountOfDucks = 1;
+        int ducksToAdd = 5;
+
+        String unitPriceFromCatalogue;
+        CataloguePage.addDucksToCart(driver, RED, String.valueOf(initialAmountOfDucks));
+        unitPriceFromCatalogue = CataloguePage.getUnitPrice(driver);
+        CataloguePage.goToCartPage(driver);
+        CartPage.clickIncreaseArrowGivenTimes(driver, ducksToAdd);
+        CartPage.updateButtonClick(driver);
+        Assert.assertEquals(driver.findElement(LocatorHelper.getLocator("CartPage.inputField")).getAttribute("value"),
```
[the remark this answers is not in the document]
Comment (andrej-khatkevich) [reply RubberDucksTest/src/test/java/BinTest.java:86]: Fixed

[the remark this answers is not in the document]
Comment (andrej-khatkevich) [reply RubberDucksTest/src/test/java/BinTest.java:86]: Yea, thank you. Moved this code to separate method in CartPage

File: RubberDucksTest/src/test/java/BinTest.java [hunk 7]
```diff
@@ -1,13 +1,114 @@
...
+     * Test case #13
+     */
+    @Description("Method adds 1 duck to bin. Then decrease it by " +
+            "clicking up arrow 5 times. Checks " +
+            "validation works well, input field stays '0' and ducks removed after update clicked")
+    @Test
+    public void decreaseElementsLessThanZeroByClickingArrowsTest() {
+        CataloguePage.addDucksToCart(driver, GREEN, "1");
+        CataloguePage.goToCartPage(driver);
+        Assert.assertEquals(driver.findElement(LocatorHelper.getLocator("CartPage.inputField")).getAttribute("value"),"1");
```
[the remark this answers is not in the document]
Comment (andrej-khatkevich) [reply RubberDucksTest/src/test/java/BinTest.java:108]: Fixed

File: RubberDucksTest/src/test/java/mavenizer/helpers/OrderSummaryRecord.java [hunk 8]
```diff
@@ -0,0 +1,43 @@
+package mavenizer.helpers;
+
+public class OrderSummaryRecord {
```
[the remark this answers is not in the document]
Comment (andrej-khatkevich) [reply RubberDucksTest/src/test/java/mavenizer/helpers/OrderSummaryRecord.java:3]: Moved to dto package

File: RubberDucksTest/src/test/java/mavenizer/helpers/OrderSummaryTableMapper.java [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+
+    public List<OrderSummaryRecord> getOrderSummaryRecords() {
+        return orderSummaryRecords;
+    }
+
+    @Override
+    public String toString() {
+        StringBuilder builder = new StringBuilder();
+        for (OrderSummaryRecord record : orderSummaryRecords) {
+            builder.append(record.getQuantity()).append(" : ").append(record.getProduct()).append(" : ").append(record.getSku())
```
[the remark this answers is not in the document]
Comment (andrej-khatkevich) [reply RubberDucksTest/src/test/java/mavenizer/helpers/OrderSummaryTableMapper.java:35]: Ok

File: RubberDucksTest/src/test/java/mavenizer/helpers/StringHelper.java [hunk 10]
```diff
@@ -0,0 +1,15 @@
+package mavenizer.helpers;
+
+import java.text.NumberFormat;
+import java.util.Formatter;
+
+public class StringHelper {
+    public static String calculateTotals(String unitPrice, int quantity) {
```
[the remark this answers is not in the document]
Comment (andrej-khatkevich) [reply RubberDucksTest/src/test/java/mavenizer/helpers/StringHelper.java:7]: Thank you

File: RubberDucksTest/src/test/java/mavenizer/staticPO/CartPage.java [hunk 11]
```diff
@@ -1,4 +1,91 @@
...
+import org.openqa.selenium.*;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+
+import java.time.Duration;
+import java.util.List;
+
 public class CartPage {
+    @Step("Return links to all categories in the bin")
+    public static List<WebElement> getCategoriesLiList(WebDriver driver) {
```
[the remark this answers is not in the document]
Comment (andrej-khatkevich) [reply RubberDucksTest/src/test/java/mavenizer/staticPO/CartPage.java:16]: I meant tag <li>. Renamed to getCategoriesLinksList.
I moved this method to CartPage cause I need refresh all links in cycle.
Clicking 'Remove' button completely changes DOM and old links stays invalid :(
Thare are another methods with same "get element" aproach like getRemoveButtonByIndex for catching link dynamically, or getOrderSummaryTable.. What is the right place for these methods? Another package?

File: RubberDucksTest/src/test/java/mavenizer/staticPO/CataloguePage.java (hunk 1 above)
Comment (andrej-khatkevich) [reply RubberDucksTest/src/test/java/mavenizer/staticPO/CataloguePage.java:34]: Fixed

**Review by Fulcrum81 — APPROVED** [review]

**Comment by QAEvgenyZarytsky** [comment]
Checked! Everything works correctly!

**Review by QAEvgenyZarytsky — APPROVED** [review]
It's good job!

**Comment by andrej-khatkevich** [comment]
Thanks for All

**Pull request merged**